pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -233,6 +233,18 @@ def auto_reduce_results_on_epoch_end(self) -> None:
 
         self.has_reduced = True
 
+    def reset(self):
+        """
+        Call at the end of epoch to reset Result objects
+        """
+        for dl_idx in range(self.num_dataloaders):
+            epoch_metrics = self._internals[dl_idx] if not self.has_reduced else self._internals_reduced[dl_idx]
+            if self._internal_type == ResultStoreType.INSIDE_BATCH_TRAIN_LOOP:
+                for opt_idx in list(epoch_metrics):
+                    epoch_metrics[opt_idx].reset()
```

Comment:
could you explain this check?

Reply:
I am not completely sure about what is going on myself, but apparently the `self._internal_type` is equal to `ResultStoreType.INSIDE_BATCH_TRAIN_LOOP` even when we are at epoch end for all training metrics. `self._internal_type` is equal to `ResultStoreType.OUTSIDE_BATCH_TRAIN_LOOP` for validation metrics (atleast when this `reset` function is called).